### DISCUSSION TITLE
Prisma adapter + MongoDB connector fails on Google signin

### DISCUSSION THREAD

**Discussion by ri5t3ai** [nextauthjs/next-auth#3851] [Help] [answered]
### Title

Prisma adapter + MongoDB connector fails on signin

### How to reproduce ☕️

I'm running the boilerplate NextAuth setup with the Prisma adapter connecting to a MongoDB Atlas instance . On Google signin the following error occurs which I tried resolving by changing the account model with no success. 

### Your question/bug report 📓

```
[next-auth][error][OAUTH_CALLBACK_HANDLER_ERROR] 
https://next-auth.js.org/errors#oauth_callback_handler_error

 Error:
Invalid `p.account.findUnique()` invocation in
C:\~\node_modules\@next-auth\prisma-adapter\dist\index.js:11:45

   8 getUserByEmail: (email) => p.user.findUnique({ where: { email } }),
   9 async getUserByAccount(provider_providerAccountId) {
  10     var _a;
→ 11     const account = await p.account.findUnique({
           where: {
             provider_providerAccountId: {
             ~~~~~~~~~~~~~~~~~~~~~~~~~~
               providerAccountId: '00000000000000000',
               provider: 'google'
             }
           },
           select: {
             user: true
           }
         })

Unknown arg `provider_providerAccountId` in where.provider_providerAccountId for type AccountWhereUniqueInput. Available args:

type AccountWhereUniqueInput {
  id?: String
  providerAccountId?: String
}
```

is there any way I can work around this error or perhaps I am missing something major?

Thank you!

### Contributing  🙌🏽

No, I'm afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
We have a separate MongoDB adapter, you could try that: https://next-auth.js.org/adapters/mongodb

A reproduction might be useful in any case.

**Comment by ri5t3ai** [comment]
I'd like to use Prisma for the models - I'll setup a reproduction in any case. Thanks!

**Comment by ndom91** [comment, marked as answer]
@ri5t3ai I did some digging through the Prisma docs, and it seems in the case of MongoDB, you must create this @@unique compound index yourself - Prisma can't/doesn't (not sure) do it in Mongo yet. This `provider__providerAccountId` is a manually specified `@@unique` constraint in our `Accounts` model in the Prisma schema.

See: https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#mongodb-4

**Comment by ri5t3ai** [comment]
Ok. I see what you mean - I'll post a solution in any case here once I have it. Really appreciate you looking into it!

**Comment by ndom91** [comment]
`nextauthjs/next-auth` has been migrated to a monorepository. The adapters code can now be found there under `packages/adapter-*`. Thanks for your interest in the project!

**Comment by avetalia** [comment]
In my case helped to run `npx prisma generate`.

I had the same issue, schema already contained `@@unique` and all fields were up to date with the latest documentation. I was assuming that re-running NextJs13 project will build all that necessary, but seems Prisma Client needs to be generated manually after changes in the model. https://authjs.dev/reference/adapter/prisma#generating-the-prisma-client
